hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/bulkinsert/RDDBucketIndexPartitioner.java

```diff
@@ -87,7 +87,7 @@ public JavaRDD<HoodieRecord<T>> doPartition(JavaRDD<HoodieRecord<T>> records, Pa
    */
   private JavaRDD<HoodieRecord<T>> doPartitionAndCustomColumnSort(JavaRDD<HoodieRecord<T>> records, Partitioner partitioner) {
     final String[] sortColumns = sortColumnNames;
-    final SerializableSchema schema = new SerializableSchema(HoodieAvroUtils.addMetadataFields((new Schema.Parser().parse(table.getConfig().getSchema()))));
+    final SerializableSchema schema = new SerializableSchema(HoodieSchemaUtils.addMetadataFields((HoodieSchema.parse(table.getConfig().getSchema()))).getAvroSchema());
```

Comment:
HoodieSchema is serializable, we do not need to wrap it around `SerializableSchema` again.